core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/DataSynchronizer.java

```diff
@@ -682,266 +706,166 @@ private void syncRemoteToLocal() throws InterruptedException {
    */
   @CheckReturnValue
   private @Nullable
-  SyncWriteModelContainer syncRemoteChangeEventToLocal(
+  LocalSyncWriteModelContainer syncRemoteChangeEventToLocal(
```

Comment:
[minor] The cyclomatic complexity of this method is fairly high at this point. Since breaking this method/class up is out of scope for now, I would really appreciate more line-by-line comments (akin to what Adam had previously).